elfi/methods.py

```diff
@@ -438,17 +442,19 @@ def sample(self, n_samples, n_populations, schedule):
             # TODO: make weights 2d as well
             weights[:] = np.concatenate(weights_t)[:n_samples]
 
-        return dict(samples=samples,
-                    distances=distances,
-                    weights=weights,
-                    threshold=threshold,
-                    n_sim=n_sim,
-                    accept_rate=accept_rate,
-                    samples_history=samples_history,
-                    distances_history=distances_history,
-                    weights_history=weights_history,
-                    threshold_history=threshold_history,
-                    accept_rate_history=accept_rate_history)
+        result = Result(OrderedDict(zip([n.name for n in self.parameter_nodes], samples)),
```

Comment:
I think I will change this back to taking samples as list and then constructing the OrderedDict within constructor.